server/src/services/asset-media.service.ts

```diff
@@ -208,10 +208,18 @@ export class AssetMediaService extends BaseService {
     const asset = await this.findOrFail(id);
     const size = dto.size ?? AssetMediaSize.THUMBNAIL;
 
-    const { thumbnailFile, previewFile } = getAssetFiles(asset.files);
+    const { thumbnailFile, previewFile, fullsizeFile } = getAssetFiles(asset.files);
     let filepath = previewFile?.path;
     if (size === AssetMediaSize.THUMBNAIL && thumbnailFile) {
       filepath = thumbnailFile.path;
+    } else if (size === AssetMediaSize.FULLSIZE) {
+      await this.requireAccess({ auth, permission: Permission.ASSET_DOWNLOAD, ids: [id] });
+      // eslint-disable-next-line unicorn/prefer-ternary
+      if (mimeTypes.isWebSupportedImage(asset.originalPath)) {
+        filepath = asset.originalPath;
+      } else {
+        filepath = fullsizeFile?.path ?? previewFile?.path;
```

Comment:
I wish to 302 redirect to original/preview path in the case when fullsize file is not available, so that we can make better use of caching, but I didn't find a good way to do the redirect either in this service or the controller/sendFile() that calls it. Any suggestions?

Reply:
You can use `res.redirect` in the controller to redirect to the other endpoint. Maybe the service can return null or a boolean to indicate whether the controller should redirect?

Reply:
implemented in https://github.com/immich-app/immich/pull/14446/commits/132e23b058418d5816970483c4b93739b1030d59